app/src/main/java/a3/com/convo/activities/LoginActivity.java

```diff
@@ -189,22 +217,38 @@ public void done(ParseException e) {
     }
 
     protected void getFriendsOnApp(AccessToken access_token) {
-        // TODO: add the friends' object IDs not facebook ids
         GraphRequest request = GraphRequest.newMyFriendsRequest(
                 access_token,
                 new GraphRequest.GraphJSONArrayCallback() {
                     @Override
                     public void onCompleted(JSONArray friends, GraphResponse response) {
                         try {
-                            ParseUser user = ParseUser.getCurrentUser();
+                            final ParseUser user = ParseUser.getCurrentUser();
                             // initialize empty likes array
-                            user.put("friends", new ArrayList<String>());
+                            user.put(FRIENDS, new ArrayList<String>());
                             for (int i = 0; i < friends.length(); i++) {
                                 JSONObject friend = friends.optJSONObject(i);
-                                String name = friend.optString("name");
-                                String id = friend.optString("id");
-                                user.add("friends", id);
-                                user.saveInBackground();
+                                String name = friend.optString(NAME);
+                                final String id = friend.optString(ID);
+                                ParseQuery<ParseUser> query = ParseUser.getQuery();
+                                query.whereEqualTo(USERNAME, id);
+                                query.findInBackground(new FindCallback<ParseUser>() {
+                                    @Override
+                                    public void done(List<ParseUser> objects, ParseException e) {
+                                        if (e == null) {
+                                            assert objects.size() == 1;
```

Comment:
if this condition is false, does this crash the app? If so, can we instead use if condition? We should be handling all kinds of data being sent to client